src/main/js/wiki/components/PageHeader.tsx

```diff
@@ -206,31 +196,67 @@ class PageHeader extends React.Component<Props, State> {
       />
     );
 
-    const branchDropdown = (
-      <BranchDropdown
-        page={page}
-        repository={this.props.wiki.repository}
-        pushBranchStateFunction={pushBranchStateFunction}
-        branch={branch}
-        branches={branches}
+    let settingsButton;
+    let branchDropdown;
+    if (!inSettings) {
+      settingsButton = (
+        <ActionButton
+          glyphicon="glyphicon-cog"
+          type="menu"
+          onClick={() => {
+            const { repository, branch } = wiki;
+            const wikiPath = `/${repository}/${branch}/`;
+            this.props.history.push(wikiPath + "settings");
+          }}
+          i18nKey="page-header_settings"
+        />
+      );
+    }
+
+    if (page && !inSettings && !inEdit) {
+      branchDropdown = (
+        <BranchDropdown
+          path={page.path}
+          repository={this.props.wiki.repository}
+          pushBranchStateFunction={pushBranchStateFunction}
+          branch={branch}
+          branches={branches}
+        />
+      );
+    }
+
+    const createForm = (
+      <PageNameForm
+        show={this.state.showCreateForm}
+        onOk={this.onOkCreate}
+        onAbortClick={this.onAbortCreateClick}
+        labelPrefix="create"
+        directory={wiki.directory}
       />
     );
 
+    const createButton = (
```

Comment:
use the hal link here

Reply:
There is no page property available for create even on an existing page. Using the same constructs like `page._links.create` will lead to an error or (when avoiding non-existing page props) to a missing create button.

I'd rather have it this way than rebuilding the whole backend.